### PR TITLE
🐛 Bump CAPI to v1.0.4

### LINKED ISSUE #1132: clusterctl in release-0.5 CI fails with cert-manager error
/kind bug

e.g. https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-openstack/1130/pull-cluster-api-provider-openstack-e2e-test/1490695117076434944/build-log.txt

```
INFO: clusterctl init --core cluster-api --bootstrap kubeadm --control-plane kubeadm --infrastructure openstack
...
action failed after 10 attempts: failed to get cert-manager object /, Kind=, /: Object 'Kind' is missing in 'unstructured object has no kind'
  occurred
```

Issue is [documented upstream](https://cluster-api.sigs.k8s.io/user/troubleshooting.html#failed-clusterctl-init---failed-to-get-cert-manager-object) with fix backported to 1.0.x branch.

### PR DESCRIPTION
**Pull request by mdbooth**
This fixes an issue preventing CI running against release-0.5.

Fixes #1132

/hold

### CODE REVIEW

**Comment by apricote** [comment]
I think this upgrade is also needed in `main` to fix the same issue. In that case it might be cleaner to upgrade in `main` first, and then backport the fix to `release-0.5`

**Comment by mdbooth** [comment]
> I think this upgrade is also needed in `main` to fix the same issue. In that case it might be cleaner to upgrade in `main` first, and then backport the fix to `release-0.5`

Thanks! I hadn't even thought to look if main was also affected 🤦‍♂️

File: go.mod [hunk 1]
```diff
@@ -20,9 +20,9 @@ require (
...
 	k8s.io/klog/v2 v2.9.0
 	k8s.io/utils v0.0.0-20210930125809-cb0fa318a74b
-	sigs.k8s.io/cluster-api v1.0.1-0.20211028151834-d72fd59c8483
-	sigs.k8s.io/cluster-api/test v1.0.1-0.20211028151834-d72fd59c8483
-	sigs.k8s.io/controller-runtime v0.10.3-0.20211011182302-43ea648ec318
+	sigs.k8s.io/cluster-api v1.0.4
+	sigs.k8s.io/cluster-api/test v1.0.4
+	sigs.k8s.io/controller-runtime v0.10.3
 	sigs.k8s.io/yaml v1.3.0
 )
```
Comment (sbueringer) [inline go.mod:28]: one line below should be changed too

**Comment by chrischdi** [comment]
> I think this upgrade is also needed in `main` to fix the same issue. In that case it might be cleaner to upgrade in `main` first, and then backport the fix to `release-0.5`

for reference: looks like it was bumped on `main` via: #1137 :-)

/lgtm

CI issue looks unrelated:

/test pull-cluster-api-provider-openstack-e2e-test

**Comment by mdbooth** [comment]
Closed in favour of https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1169

**Pull request closed without merging**